wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -757,8 +764,12 @@ public Revision commit(CommitParams params) throws GitException {
       boolean isGerritSupportConfigured =
           gerritSupportConfigValue != null ? Boolean.valueOf(gerritSupportConfigValue) : false;
       commitCommand.setInsertChangeId(isGerritSupportConfigured);
+      String repositoryPath = getRepository().getDirectory().getPath();
+      COMMITTING_REPOSITORIES.add(repositoryPath);
       RevCommit result = commitCommand.call();
+      COMMITTING_REPOSITORIES.remove(repositoryPath);
```

Comment:
Let's clean up committing status on final block only.

Reply:
In that case status execution will throw exception:
https://github.com/eclipse/che/blob/1c05caf0c1f1a8e76a1fd0777d614967574e9da9/wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java#L769-L772

Reply:
BTW commit is already finished on `commitCommand.call()`